securedrop/journalist_app/api2/__init__.py

```diff
@@ -14,18 +15,57 @@
 PREFIX_MAX_LEN = inspect(Source).columns["uuid"].type.length
 
 
-def all_sources() -> Query:
-    """
-    Return a base query for all ``Sources`` with eager loading of their metadata
-    and collections.
-    """
+# Used by `{Submission,Reply}_query_options()` when called directly rather than
+# via `Source_query_options()`.
+LOADER_BASE = {
+    Submission: Load(Submission),
+    Reply: Load(Reply),
+}
+
+
+def Submission_query_options(base: Load = LOADER_BASE[Submission]) -> Tuple:
+    configure_mappers()
+    return (
+        base.joinedload(Submission.source),  # type: ignore[attr-defined]
+        base.joinedload(Submission.seen_files).joinedload(SeenFile.journalist),  # type: ignore[attr-defined]
+        base.joinedload(Submission.seen_messages).joinedload(SeenMessage.journalist),  # type: ignore[attr-defined]
+    )
+
+
+def Reply_query_options(base: Load = LOADER_BASE[Reply]) -> Tuple:
+    configure_mappers()
     return (
-        Source.query.options(joinedload(Source.star))
-        .options(joinedload(Source.submissions))
-        .options(joinedload(Source.replies))
+        base.joinedload(Reply.source),  # type: ignore[attr-defined]
+        base.joinedload(Reply.journalist),  # type: ignore[attr-defined]
+        base.joinedload(Reply.seen_replies).joinedload(SeenReply.journalist),  # type: ignore[attr-defined]
     )
 
 
+def Source_query_options() -> Tuple:
+    configure_mappers()
+    return (
+        joinedload(Source.star),
+        *Submission_query_options(joinedload(Source.submissions)),
+        *Reply_query_options(joinedload(Source.replies)),
+    )
+
+
+EAGER_BUNDLES = {
+    Source: lambda: Source_query_options(),
```

Comment:
I don't think you need the lambda, I think something like this should work?

```suggestion
    Source: Source_query_options,
```

Reply:
Fixed in bf39c6e74db1c727d6c6955795d8eb5739dc3732.